### PR TITLE
Clean up text drawer compression issues

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fixes several problems with `vertical_compression` in the text circuit drawer.

### Details and comments

There is an option for the text circuit drawer called `vertical_compression` that allows a user to choose how much vertical spacing there is between the horizontal lines. If `'high` is chosen, then only 1 line will display between a pair of wires and in certain cases it's possible characters will overwrite other characters or drawing elements. If `low` is chosen, 2 lines will be used between every pair of lines and no overwriting will occur.

If `medium`, which is the default, is chosen, the compression starts out the same as `high`, but the drawer then tries to detect areas where overwriting would occur and expands the vertical to 2 lines where that would happen. The detection of these areas was not working consistently.

`low` compression was also causing connections between a gate and a control to be broken. Finally there were areas where connections for conditional gates were breaking. These drawings show 2 of the problems followed by a corrected drawing.

```       a             
qr_0: ─░─────────────
       ░       ┌───┐ 
qr_1: ─░───■───┤ X ├─
       ░   │   Bar─1 
qr_2: ─░───┼─────░───
       ░ ┌─┴─┐   ░   
qr_3: ───┤ X ├───░───
         └───┘   ░   

       a             
qr_0: ─░─────────────
       ░       ┌───┐ 
qr_1: ─░───■───┤ X ├─
       ░   │   └───┘ 
       ░   │   Bar 1 
qr_2: ─░───┼─────░───
       ░ ┌─┴─┐   ░   
qr_3: ───┤ X ├───░───
         └───┘   ░   

qr_0: ────────────
            ┌───┐ 
qr_1: ──■───┤ X ├─
        │   └───┘ 
            Bar 1 
qr_2: ────────░───
      ┌─┴─┐   ░   
qr_3: ┤ X ├───░───
      └───┘   ░   

qr_0: ────────────
            ┌───┐ 
qr_1: ──■───┤ X ├─
        │   └───┘ 
        │   Bar 1 
qr_2: ──┼─────░───
      ┌─┴─┐   ░   
qr_3: ┤ X ├───░───
      └───┘   ░   
```

### CODE REVIEW

File: test/python/visualization/test_circuit_text_drawer.py [hunk 1]
```diff
@@ -1688,22 +1688,22 @@ def test_control_gate_label_with_cond_2_med(self):
...
-                "   c: 0 ═══■════",
-                "          0x1   ",
+                "          ┌──────┐ ",
+                "q_0: |0>──┤ my h ├─",
+                "          └──┬───┘ ",
+                "q_1: |0>─────■─────",
+                "         my ctrl-h ",
+                "             ║     ",
+                "   c: 0 ═════■═════",
+                "            0x1    ",
```
Comment (jakelishman) [inline test/python/visualization/test_circuit_text_drawer.py:1698]: Can we revert this test case?  The behaviour that appears with this PR and this test is probably something we want to make a positive decision on, and potentially test both it and your new form (with the space not aligned).

For clarity: with this PR, the original test produces
```text
        ┌──────┐
q_0: |0>┤ my h ├
        └──┬───┘
q_1: |0>───■────
         my║ch
   c: 0 ═══■════
          0x1
```
instead.  I feel like that (^) behaviour is probably actually suitable for `vertical_compression=medium` - it compresses vertically when there's a "space" so nothing visible gets erased, whereas "high" _always_ compresses and "low" never does.

Perhaps leave the test as-is, and add your new one as a separate case?

**Review by jakelishman — COMMENTED** [review]
Thanks Edwin!  One minor comment about the tests, but I think the new behaviour is probably a pretty reasonable default.

File: test/python/visualization/test_circuit_text_drawer.py (hunk 1 above)
Comment (enavarro51) [reply test/python/visualization/test_circuit_text_drawer.py:1698]: Ha, I was hoping nobody would notice that. It works visually either way and it's probably a pretty rare case. I'll add the old test back with the new one.

**Review by jakelishman — APPROVED** [review]
Looks great, thanks for the changes!

**Pull request merged**